redux/actions/workspace.js

```diff
@@ -0,0 +1,96 @@
+var firebaseApp = require('../../js/Firebase.jsx');
+require('isomorphic-fetch');
+
+const setCurrentPlace = (placeId) => {
+    return {
+        type: 'SET_CURRENT_PLACE',
+        placeId
+    }
+}
+
+
+const fetchWorkspaceData = (filterParams) => {
+    return (dispatch) => {
+        let placesArray = []
+        let workspacesRef = firebaseApp.ref('/workspaces/');
+        workspacesRef.once('value').then(snapshot => {
+        const data = snapshot.val();
+        const workspaces = Object.keys(data).map(key => data[key]);
+            dispatch(fetchMapData(workspaces));
+        });
+    }
```

Comment:
Should return `Promise.all(workspaces)` so you can chain async actions in components.